runner/full.go

```diff
@@ -17,25 +19,71 @@ import (
 	"github.com/pganalyze/collector/util"
 )
 
-func collectDiffAndSubmit(server state.Server, globalCollectionOpts state.CollectionOpts, logger *util.Logger) (state.PersistedState, error) {
+const MinSupportedLogMinDurationStatement = 100
+
+func validateLogCollectionConfig(server *state.Server, settings []state.PostgresSetting) (bool, string) {
```

Comment:
I'm wondering if we should move this to something like `logs/validate_config.go` instead, to keep it colocated with most of the other log handling code?

Reply:
Hmm, should we integrate the `log_line_prefix` checks into this validation?

Reply:
On second thought, I think I'm going to leave this for now--that check uses a different mechanism, and I think this patch is probably big enough as is. We may want to revisit this but let's land this first. I did move the check logic to the logs package as you suggest.